cmd/zpool/zpool_main.c

```diff
@@ -7253,15 +7253,71 @@ print_error_log(zpool_handle_t *zhp)
 	elem = NULL;
 	while ((elem = nvlist_next_nvpair(nverrlist, elem)) != NULL) {
 		nvlist_t *nv;
-		uint64_t dsobj, obj;
-
-		verify(nvpair_value_nvlist(elem, &nv) == 0);
-		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_DATASET,
-		    &dsobj) == 0);
-		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_OBJECT,
-		    &obj) == 0);
+		uint64_t dsobj, obj, data_block_size, indirect_block_size;
+		uint64_t *block_ids;
+		int64_t *indirect_levels;
+		unsigned int error_count_blkids;
+		unsigned int error_count_levels;
+
+
+		nv = fnvpair_value_nvlist(elem);
+		dsobj = fnvlist_lookup_uint64(nv, ZPOOL_ERR_DATASET);
+		obj = fnvlist_lookup_uint64(nv, ZPOOL_ERR_OBJECT);
+		block_ids = fnvlist_lookup_uint64_array(nv,
+		    ZPOOL_ERR_BLOCKID, &error_count_blkids);
+		indirect_levels = fnvlist_lookup_int64_array(nv,
+		    ZPOOL_ERR_LEVEL, &error_count_levels);
+
+		ASSERT(error_count_levels == error_count_blkids);
```

Comment:
Can we use ASSERT3U here? Not sure if it's defined in this file.

Reply:
it is not defined here